R/readCao2024.R

```diff
@@ -0,0 +1,571 @@
+        "percentage of waste concrete (for road base, backfills materials and other use) with particle size <1mm (min)",
+        "percentage of waste concrete (for road base, backfills materials and other use) with particle size 1-10mm (min)",
+        "percentage of waste concrete (for road base, backfills materials and other use) with particle size 10-30mm (min)",
+        "percentage of waste concrete (for road base, backfills materials and other use) with particle size 30-53mm (min)",
+        "percentage of waste concrete (landfill, dumped and stacking) with particle size <10mm (min)",
+        "percentage of waste concrete (landfill, dumped and stacking) with particle size 10-30mm (min)",
+        "percentage of waste concrete (landfill, dumped and stacking) with particle size 30-50mm (min)",
+        "percentage of waste concrete (landfill, dumped and stacking) with particle size >50mm (min)",
+        "percentage of waste concrete (asphalt concrete) with particle size <5mm (min)",
+        "percentage of waste concrete (asphalt concrete) with particle size 5-10mm (min)",
+        "percentage of waste concrete (asphalt concrete) with particle size 10-20mm (min)",
+        "percentage of waste concrete (asphalt concrete) with particle size 20-40mm (min)"
+      ),
+      dim_members = list(
+        "Concrete Waste Type" = c("new concrete", "aggregates", "landfill", "asphalt"),
+        "Particle Size" = c("A", "B", "C", "D")
+      ),
+      dim = c("Concrete Waste Type", "Particle Size"),
+      normalize = TRUE,
+      groups = c(rep("new concrete", 4), rep("aggregates", 4), rep("landfill", 4), rep("asphalt", 4))
+    ),
+    CKD_landfill_share = list(
+      long_names = c("percentage of CKD sent to landfill (distribution)")
+    ),
+    clinker_cao_content = list(
+      long_names = c("average CaO content of clinker in cement (fCaO) (distribution)")
+    ),
+    CKD_cao_content = list(
+      long_names = c("CaO content in CKD (distribution)")
+    )
+  )
+
+  path <- file.path("v1", "data_cement_GAS_EoL_MISO_9regions.xlsx")
+  data <- suppressMessages(readxl::read_xlsx(path, sheet = "Uptake"))
+  data <- head(data, 11) # remove rows after row 11
+  data <- data[-1, , drop = FALSE] # remove row 2
+
+  # parameters for data gathering
+  normalize <- FALSE
+  dim_members <- NULL
+  dim <- NULL
+  fraction_mean <- FALSE
+  groups <- NULL
+
+  spec <- cao2024_specs[[subtype]]
+
+  if (is.null(spec)) {
+    # this is not used.
+    long_names <- c("ratio of CO2 element to CaO (Mr)")
+      } else {
+    long_names <- spec$long_names
+    if (!is.null(spec$dim_members)) dim_members <- spec$dim_members
+    if (!is.null(spec$dim)) dim <- spec$dim
+    normalize <- isTRUE(spec$normalize)
+    fraction_mean <- isTRUE(spec$fraction_mean)
+    if (!is.null(spec$groups)) groups <- spec$groups
+  }
+
+  x <- calculate_means(data, long_names, dim_members, dim,
```

Comment:
You forgot to update all the function calls after converting these functions to tool functions.